lib/SlackBot.js

```diff
@@ -207,19 +207,24 @@ function Slackbot(configuration) {
 
         if (slack_botkit.config && slack_botkit.config.webserver && slack_botkit.config.webserver.static_dir)
             static_dir = slack_botkit.config.webserver.static_dir;
-
-        slack_botkit.config.port = port;
+        
+        if (slack_botkit.config) {
+            slack_botkit.config.port = port;
+            slack_botkit.config.ip = slack_botkit.config.ip || '127.0.0.1';
+        }
+        
+        var ip = slack_botkit.config? slack_botkit.config.ip : '127.0.0.1';
 
         slack_botkit.webserver = express();
         slack_botkit.webserver.use(bodyParser.json());
         slack_botkit.webserver.use(bodyParser.urlencoded({ extended: true }));
         slack_botkit.webserver.use(express.static(static_dir));
 
         var server = slack_botkit.webserver.listen(
-            slack_botkit.config.port,
+            port,
```

Comment:
Can you clarify or rephrase your sentence please? You imply in an earlier diff comment that we cannot assume that slack_botkit.config is defined. Given this, and the fact that we enforce that port is defined earlier in lines 199-201 and will overwrite anything in slack_botkit.config.port, the use of port should not result in any change in behaviour.